### ISSUE TITLE
Unexpected spaces when ascii = False

### ISSUE THREAD

**Issue by bamattsson** [tqdm/tqdm#259]
Unsure if this is a unicode error on my side or not. 

However, when executing the following code:

```
from tqdm import tqdm
from time import sleep

for i in tqdm(range(100), ascii = True):
    sleep(0.01)
```

everything works as expected:
![asciitrue](https://cloud.githubusercontent.com/assets/10145139/18280457/584cbdc4-7459-11e6-8230-4a441cead8c6.gif)

But when setting ascii = False (or removing it) it looks like this:
![asciifalse](https://cloud.githubusercontent.com/assets/10145139/18280506/83e8851c-7459-11e6-9f87-08e104edcfa3.gif)

I should mention that it works fine in IPython. I also have a swedish keyboard and my locale looks like this:
```
LANG="sv_SE.UTF-8"
LC_COLLATE="sv_SE.UTF-8"
LC_CTYPE="sv_SE.UTF-8"
LC_MESSAGES="sv_SE.UTF-8"
LC_MONETARY="sv_SE.UTF-8"
LC_NUMERIC="sv_SE.UTF-8"
LC_TIME="sv_SE.UTF-8"
LC_ALL=
```

I have no idea what happens, but it seems like it's inserting some extra white spaces between the smooth blocks with unicode

And thanks for great work with tqdm, I will sure use it a lot!

**Comment by lrq3000** [comment]
Hello @bamattsson,

Unicode support is quite heterogenous currently: some environments implement it very well while others are more slippery.

It seems to me that your console simply does either not implement correctly unicode characters, or the font used by default (or did you choose it?) is not handling unicode well. If the latter, this could maybe be fixed by using another unicode font.

**Comment by bamattsson** [comment]
@lrq3000 Yep, it is due to settings I have. The issue was that in the terminal.app (mac) I had checked the box in Preferences menu > Settings > Advanced tab > Unicode East Asian Ambiguous characters are wide. (the rest are standard settings)

Basically the unicode characters in the progress bar have what's called an ambiguous width, which means that you have to infer from the context how to interpret them. That's why the mac terminal has that choice. So it is not due to a slippery implementation in the unicode font.

I guess one way to fix it is to find another set of characters that are not ambiguous. Or you could try to infer how the output prints the ambiguous characters (not sure if it is possible) and use ascii in the case they are UEAA wide. If it's worth fixing, maybe so few east asians are using tqdm that it doesn't matter.

Some links if you want to read more:
http://www.unicode.org/reports/tr11/
https://github.com/bpython/bpython/issues/254

**Comment by lrq3000** [comment]
Thank you @bamattsson for tracking down the root of the bug, this is greatly helpful!

Indeed auto-detecting this console's parameter would be great but I'm not sure if it is possible, we'll have to look into that. However, I think it's useless to try to use another set of unicode characters, because I guess that any other "smooth" characters will have the same issue (because that's why we are using them: they have a variable size), but anyway it will be soon possible for users to override the set of characters the bar uses with PRs #223 and #227.

**Comment by casperdcl** [comment]
Yes, I hava a similar issue on my phone when running tqdm in the terminal - everything works as expected except the since full-block unicode character, which is a few pixels wider than a normal character so things start going off screen. interestingly, wrap-around (when the things off screen go onto a new line) isn't triggered, probably because the terminal knows that the number of characters printed equals the terminal width (it doesn't know some are "wide" chars)
